src/lib/createVNode.js

```diff
@@ -3,8 +3,28 @@
 // 2. 반환값은 { type, props, children } 형태의 객체여야 합니다.
 // 3. children은 평탄화(flat)되어야 하며, falsy 값은 필터링되어야 합니다.
 // 4. Infinity를 사용하여 모든 깊이의 배열을 평탄화하세요.
-
+function flatten(arr) {
+  return arr.reduce((acc, cur) => {
+    if(Array.isArray(cur))
+      return [...acc, ...flatten(cur)];
+    return [...acc, cur];
+  }, []);
+}
 export function createVNode(type, props, ...children) {
   // 여기에 구현하세요
-  return {}
+  
+  const $children = [];
+  children.map((child) => {
+    if(typeof child === 'string' || typeof child === 'number')
+      $children.push(child);
+    else if(Array.isArray(child))
+      $children.push(...flatten(child));
+    else if(child)
+      $children.push(child);
+  });
+  return {
+    type,
+    props,
+    children: $children
```

Comment:
```suggestion
  children.map((child) => {
    if(typeof child === 'string' || typeof child === 'number')
      $children.push(child);
    else if(Array.isArray(child))
      $children.push(...flatten(child));
    else if(child)
      $children.push(child);
  });
  return {
    type,
    props,
    children: children.flat(infinity)
```
이런식으로 표현하는 방법도 있답니다!

Reply:
혹은 현재 로직을 그대로 활용한다고 치면, 이렇게 할 수 있어요!
```tsx
function flatten(arr) {
  return arr.reduce((acc, cur) => acc.concat(Array.isArray(cur) ? flatten(cur) : cur), []);
}

export function createVNode(type, props, ...children) {
  return {
    type,
    props,
    children: flatten(children).filter(Boolean),
  }

```